src/polynomial/mod.rs

```diff
@@ -72,6 +72,8 @@ impl<F: FiniteField> Basis for Lagrange<F> {
 }
 
 impl<B: Basis, F: FiniteField, const D: usize> Polynomial<B, F, D> {
+  const CHECK: () = assert!(D.is_power_of_two());
```

Comment:
We don't need to have this as an associated constant to the implementation. Instead we can place this only on the FFT.